### ISSUE TITLE
Search results are obstructed when moving up through the list

### ISSUE THREAD

**Issue by golota60** [algolia/docsearch#1572]
## Description

hey! thanks for this awesome library.
downstream issue: https://github.com/react-bootstrap/react-bootstrap/issues/6437

This bug is is happening cause the [upstream autocomplete library]() is using [scrollIntoView and scrollIntoViewIfNeeded](https://github.com/algolia/autocomplete/blob/5ba25f62213b2721218fb34ecc9472286cb9f926/packages/autocomplete-core/src/onKeyDown.ts#L32) which don't account for the sticky header. This has to be fixed upstream(by possibly allowing to pass an offset?). I've created this issue to track the fix here once upstream makes it possible to be fixed.  

## Steps to reproduce

1. Go to https://docsearch.algolia.com/
2. Click on the search button in the upper right corner.
3. In the modal that is shown, input something so that the list gets populated with results
4. Use arrow keys to navigate to the bottom of the modal
5. Use arrow key up to go up through the list so that the list scrolls upwards
6. The results that are being scrolled into view are not fully shown due to the sticky header

**Live reproduction:**

[docsearch-bug.webm](https://user-images.githubusercontent.com/37072867/191042474-5cf4001b-ffdc-48a8-90ae-d253de919f95.webm)

## Expected behavior

Elements should scroll underneath the sticky header so that they're visible

## Environment

- OS: Linux 
- Browser: Chrome 103
- DocSearch version: 3.2.1

**Comment by golota60** [comment]
Upstream discussion: https://github.com/algolia/autocomplete/discussions/1019

**Comment by sarahdayan** [comment]
This seems to be a Chrome behavior, Firefox scrolls differently:

https://user-images.githubusercontent.com/5370675/191471876-7e191e72-2c5c-41f3-9f55-bd6459115bf5.mov

Could the [`scroll-margin-top` CSS property](https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-margin-top) help here? That could be implemented in DocSearch (cc @shortcuts).

**Comment by golota60** [comment]
Yeah, this prop seems to be fixing the issue - will create a PR shortly. Thanks!
